### DISCUSSION TITLE
Clean a left nozzle just before printing Davinci 2.0A Duo

### DISCUSSION THREAD

**Discussion by DBSDBS** [luc-github/Repetier-Firmware-0.92#383] [Q&A]
Hi,

In the original XYZ printer software, there was an integrated functionality, that makes a cleaning of the two nozzles just before the printing starts. That means: Heathing, some extrusion, cleaning (pass trough the cleaners), print a sample rectangle on the left top corner an AFTER that start the real printing.

Is there a way to repeat this procedure - some G / M codes and / or some other procedure?

Thank you in advance.

Best Regards

**Comment by luc-github** [comment]
All available GCODE are here https://github.com/luc-github/Repetier-Firmware-4-Davinci/blob/master/src/ArduinoDUE/Repetier/Repetier.ino

 `M100 T<extruderNum> clean nozzle, no extruder specified clean all available`

**Comment by DBSDBS** [comment]
Thank you!
